app/models/server.rb

```diff
@@ -228,6 +229,29 @@ def collect_supported_tests
     self.save!
   end
 
+  def check_badges
+    badges = []
+    Badge.all.each do |badge|
+      earned = true
+      badge.suites.each do |suite|
+        if not self.supported_suites.include? suite
```

Comment:
From what I understand, using "unless" rather than "if not" as you did in 237 and 242 here is more Ruby-like.